misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/model/JMethod.java

```diff
@@ -0,0 +1,181 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.bindgen.model;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+import static org.ballerinalang.bindgen.command.BindingsGenerator.getAllJavaClasses;
+import static org.ballerinalang.bindgen.command.BindingsGenerator.setClassListForLooping;
+import static org.ballerinalang.bindgen.utils.BindgenConstants.BALLERINA_RESERVED_WORDS;
+import static org.ballerinalang.bindgen.utils.BindgenConstants.BALLERINA_STRING;
+import static org.ballerinalang.bindgen.utils.BindgenConstants.BALLERINA_STRING_ARRAY;
+import static org.ballerinalang.bindgen.utils.BindgenConstants.METHOD_INTEROP_TYPE;
+import static org.ballerinalang.bindgen.utils.BindgenUtils.getBallerinaHandleType;
+import static org.ballerinalang.bindgen.utils.BindgenUtils.getBallerinaParamType;
+import static org.ballerinalang.bindgen.utils.BindgenUtils.isStaticMethod;
+
+/**
+ * Class for storing details pertaining to a specific Java method used for Ballerina bridge code generation.
```

Comment:
As a practice, we add @since annotation. Shall we add that.

Reply:
Added the since annotation.